projects/plugins/boost/app/lib/critical-css/source-providers/providers/Archive_Provider.php

```diff
@@ -92,13 +92,25 @@ public static function get_post_types() {
 			array(
 				'public'      => true,
 				'has_archive' => true,
-			)
+			),
+			'objects'
 		);
 		unset( $post_types['attachment'] );
 
 		$post_types = array_filter( $post_types, 'is_post_type_viewable' );
 
-		return apply_filters( 'jetpack_boost_critical_css_post_types', $post_types );
+		$provider_post_types_ignore_list = Singular_Post_Provider::get_ignored_post_types();
+		$provider_post_types             = array();
+		// Generate a name => name array for backwards compatibility.
+		foreach ( $post_types as $post_type ) {
+			if ( in_array( $post_type->name, $provider_post_types_ignore_list, true ) ) {
+				continue;
+			}
+
+			$provider_post_types[ $post_type->name ] = $post_type->name;
+		}
+
+		return apply_filters( 'jetpack_boost_critical_css_post_types_archives', $provider_post_types, $post_types );
```

Comment:
Filter name has changed with no deprecation period. I don't have a strong opinion on this either way - but just noting that filters are public and if there's code outside our codebase relying on this filter, that code will break.

Reply:
You're right. Even though it's probably not used, I've left it in there. Though I decided to wrap the new filter around it so we have a clearer distinction between archive and post type providers.

Updated in 471a6aa.